docs/Common-Issues-for-Core-Contributors: add steps to fix failed bottle publish (#7869)
Homebrew/brew: docs/Common-Issues-for-Core-Contributors.md

## Issues

### Bottle publishes failed but the commits are correct in the git history

Follow these steps to fix this issue:

> * `git reset --hard origin/master` to reset to the latest commit before the bottle publish.

Reviewer: This reset should be done to the commit before all the commits created by `brew pr-pull`.

> * `git reset --hard origin/master` to reset to the latest commit before the bottle publish.

Reviewer 2: ```suggestion
* `git reset --hard origin/HEAD` in homebrew/core to reset to the latest commit before the bottle publish.
```

> * `git reset --hard <SHA>` in homebrew/core to reset to the commit before before all the commits created by `brew pr-pull`.

Reviewer: ```suggestion
* `git reset --hard <SHA>` in homebrew/core to reset to the commit before all the commits created by `brew pr-pull`.
```

> * `brew pr-pull <options>` to upload the right bottles.

Reviewer 2: ```suggestion
* `brew pr-pull <options>` to upload the right bottles. Add the `--warn-on-upload-failure` flag if the bottles have been partially uploaded and you're certain that the bottle checksums will match the checksums already present in the `bottle do` block of the formula.
```
